envoy/api/v2/route/route.proto

```diff
@@ -768,6 +768,17 @@ message RateLimit {
   repeated Action actions = 3 [(validate.rules).repeated .min_items = 1];
 }
 
+// [#not-implemented-hide:]
+// Specifies the int64 lower and upper bounds of a range.
+// Header route matching will be performed if the header's value lies within this range (inclusive).
+message HeaderValueRange {
+  // Low key, inclusive
+  google.protobuf.Int64Value low_key = 1;
```

Comment:
Do we need a wrapped type here? Is the idea that if it's not specified, this is an open ended range (lower is min int64 if missing, upper is max int64)?

Reply:
IMO the low and high could be "required" fields, no default values will be assumed. Being more explicit will avoid confusion. What do you say? 
Regarding using wrapped types, are there any disadvantages? New to protos here so I can change this to int64 if that is the recommended way.

Reply:
HeaderValueRange is already optional. The low and high can be required. By the way, what "key" means here.

If the comment says, lower bound and upper bound, the field name should read the same. Please avoid asking readers to correlate "lower bound" and "low key".

Reply:
+1, I would also use primitive types here. IMO default of 0 and 0 is fine.

Reply:
Please see my other comment about defining a more generic type. This type is so commonly used, we should promote it to the top level.

```
message Range {
  double start;
  double end;
}
```

Reply:
@wora top level => some common proto file? I see core/base.proto. Is that the place for such generic types?
Based on Matt's other comment, to avoid double conversions, shall we stick to int64 then?

```
message Int64Range {
  required int64 start;
  required int64 end;
}
```

Reply:
For types that are never going to change, I suggest we put them at the very top level, e.g. /envoy/type/range.proto. We can put Int64Range there first, add more if needed.